### ISSUE TITLE
Property names are missing in error text

### ISSUE THREAD

**Issue by kutyepov** [seriousme/fastify-openapi-glue#222]
From version `v2.4.1` onwards, property names are missing in the response body. This is due to a breaking change in `ajv@8`:

> property name is removed from "propertyName" keyword error message (it is still available in error.params.propertyName).
> more details [here](https://github.com/ajv-validator/ajv/releases/tag/v8.0.0)

This issue was introduced in #190 where ajv@7 was upgraded to avj@8. Specific example of missing property name can be seen in [updated test](https://github.com/seriousme/fastify-openapi-glue/pull/190/files?file-filters%5B%5D=.js) in that PR.

---
I am happy to raise a PR to fix this issue, but need some guidance on what would be the best approach to tackle it.
So far, I found an approach that works. 
If the following is added to [setValidatorCompiler](https://github.com/seriousme/fastify-openapi-glue/blob/master/index.js#L47) in index.js: 
```
  instance.setSchemaErrorFormatter(function (errors, dataVar) {
    return new Error(ajv.errorsText(errors))
  })
```
then the error text will include property name.
~~However, I am not sure if this is the right place to do so. Potentially, this might need to be addressed in fastify itself since it has its own ajv instance and therefore it might have the same issue.~~ Fastify@3 is using ajv@6, so their schemas work fine.

**Comment by seriousme** [comment]
Hi,

thanks for asking!

I think AJV will have a reason for excluding the property name in version 8.
(esp. since its mentioned explictly ;-))
My suspicion is because of potential information leakage.
The example Fastify provides for `setSchemaErrorFormatter` also aims at providing generic error texts. 

Fastify will be [migrating to AJV@8 around end Q2, 2021](https://github.com/fastify/fastify/issues/2742)  (to be released as Fastify 4) and I am hoping to let go of the custom AJV instance in the next major release (hence [my PR on ajv-formats](https://github.com/ajv-validator/ajv-formats/pull/22) ;-) ) while also supporting Fastify 4.

So even if we add it now, but it is probably gone in the next major release.

Kind regards,
Hans

**Comment by seriousme** [comment]
Closed a bit too soon, sorry :-(
Published as 2.5.1 on NPM

Thanks,
Hans

**Comment by kutyepov** [comment]
Yea, hopefully this will get fixed with fastify v4. Happy to undo this change when v4 is released.

Thank you for merging and releasing this change so quickly.
